src/Jobs/NodeChannelSuscribeJob.cs

```diff
@@ -0,0 +1,187 @@
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+using FundsManager.Data.Repositories;
+using FundsManager.Data.Repositories.Interfaces;
+using FundsManager.Services;
+using Grpc.Core;
+using Lnrpc;
+using NBitcoin.Protocol;
+using Quartz;
+using Channel = FundsManager.Data.Models.Channel;
+using Node = FundsManager.Data.Models.Node;
+
+namespace FundsManager.Jobs;
+
+/// <summary>
+/// Job for update the status of the channels
+/// </summary>
+/// <returns></returns>
+[DisallowConcurrentExecution]
+public class NodeChannelSuscribeJob : IJob
+{
+    private readonly ILogger<NodeChannelSuscribeJob> _logger;
+    private readonly ILightningService _lightningService;
+    private readonly INodeRepository _nodeRepository;
+    private readonly IChannelRepository _channelRepository;
+    private readonly ISchedulerFactory _schedulerFactory;
+    
+    public NodeChannelSuscribeJob(ILogger<NodeChannelSuscribeJob> logger, ILightningService lightningService, INodeRepository nodeRepository, IChannelRepository channelRepository, ISchedulerFactory schedulerFactory)
+    {
+        _logger = logger;
+        _lightningService = lightningService;
+        _nodeRepository = nodeRepository;
+        _channelRepository = channelRepository;
+        _schedulerFactory = schedulerFactory;
+    }
+        
+    public async Task Execute(IJobExecutionContext context)
+    {
+        var data = context.JobDetail.JobDataMap;
+        var nodeId = data.GetInt("nodeId");
+        _logger.LogInformation("Starting {JobName} for node {nodeId}... ", nameof(NodeChannelSuscribeJob), nodeId);
+        try
+        {
+            var node = await _nodeRepository.GetById(nodeId);
+            
+            if (node == null)
+            {
+                _logger.LogInformation("The node: {@Node} is no longer ready to be supported quartz jobs", node);
+                return;
+            }
+
+            var client = LightningService.CreateLightningClient(node.Endpoint);
+            var result = client.Execute(x => x.SubscribeChannelEvents(new ChannelEventSubscription(), 
+                new Metadata {
+                    {"macaroon", node.ChannelAdminMacaroon}
+                }, null, default));
+
+            while (await result.ResponseStream.MoveNext())
+            {
+                node = await _nodeRepository.GetById(nodeId);
+            
+                if (node == null)
+                {
+                    _logger.LogInformation("The node: {@Node} is no longer ready to be supported quartz jobs", node);
+                    return;
+                }
+                
+                try {
+                    var channelEventUpdate = result.ResponseStream.Current;
+
+                    if (channelEventUpdate.OpenChannel != null)
+                    {
+                        if (String.IsNullOrEmpty(channelEventUpdate.OpenChannel.CloseAddress))
```

Comment:
Is this really possible to happen?

Reply:
If the channelacceptor for some reason fails that could happen. Actually it was happening to me before I fixed the ChannelAcceptor init. It isnt usually null but an empty string.

Reply:
It also happens when you open from outside of NG a channel from two registered nodes. It detects two events, one with that field empty.